gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HivePartitionExtendedFilterFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.data.management.copy.hive;
+
+import com.typesafe.config.Config;
+/**
+ * A factory for {@link HivePartitionExtendedFilter} instances.
+ */
+public interface HivePartitionExtendedFilterFactory {
+  /**
+   * Filter requires information provided by {@link Config} like condition for filtering.
```

Comment:
Please reformat this javadoc. It is easier to explain simply what is expected of this method, and possibly what the `Confg` contains.